File: dart/821.dart
class Solution {
    int abs(int i) {
        return (i < 0) ? -i : i;
    }

    List<int> shortestToChar(String w, String c) {
        int n = w.length;
        List<int> r = [];
        List<int> p = [];
        int i = 0, k = 0;
        for (int j = 0; j < n; j++)
            if (w[j] == c)
                p.add(j);
        for (int j = 0; j < n; j++) {
            if (k + 1 < p.length && abs(p[k] - j) > abs(p[k + 1] - j))
                k++;
            r.add(abs(p[k] - j));
        }
        return r;
    }
}

File: python3/821.py
class Solution:
    def abs(self, i):
        return -i if i < 0 else i

    def shortestToChar(self, w: str, c: str) -> List[int]:
        n = len(w)
        r = []
        p = [j for j in range(n) if w[j] == c]
        k = 0
        for j in range(n):
            if k + 1 < len(p) and self.abs(p[k] - j) > self.abs(p[k + 1] - j):
                k += 1
            r.append(abs(p[k] - j))
        return r
